lib/invitations/cancelInvitation.ts

```diff
@@ -0,0 +1,52 @@
+import { prisma } from "@lib/integration/prismaConnector";
+import { InvitationNotFoundError } from "./exceptions";
+import { checkPrivileges } from "@lib/privileges";
+import { UserAbility } from "@lib/types";
+
+/**
+ * Cancel an invitation
+ *
+ * @param ability
+ * @param invitationId
+ */
+export const cancelInvitation = async (ability: UserAbility, invitationId: string) => {
+  // Retrieve the invitation
+  const invitation = await prisma.invitation.findUnique({
+    where: {
+      id: invitationId,
+    },
+    select: {
+      email: true,
+      templateId: true,
+    },
+  });
+
+  if (!invitation) {
+    throw new InvitationNotFoundError();
+  }
+
+  checkPrivileges(ability, [
+    { action: "update", subject: { type: "FormRecord", object: { id: invitation.templateId } } },
```

Comment:
In order to verify if a User has the privilege to update a Form Record we must pass in the Form Record Object, or it's similie.
In this case we would need to get the Users associated with the template and then pass them in as the `object` being verified.
```js
{ action: "update", subject: { type: "FormRecord", object: { users:[/* users array from actual Form Record */] } } }
```

Reply:
Updated to retrieve the template using getTemplateWithAssociatedUsers (using the templateId from the invitation) and then pass that in as the object https://github.com/cds-snc/platform-forms-client/pull/4386/commits/388eb61138cce0cc97bf2fc29564412a5e0b6177#diff-b24305f4af2ab739a4a431f95f5882531b0aa2127c0b11a945fc6eebb5c76d59R30-R37